### PR TITLE
use fully qualified job name for azkaban.link.attempt.url and azkaban.link.jobexec.url

### PR DESCRIPTION
**Pull request by edwinalu**

### CODE REVIEW

File: azkaban-exec-server/src/main/java/azkaban/execapp/JobRunner.java [hunk 1]
```diff
@@ -757,10 +758,10 @@ private void insertJobMetadata() {
       this.props.put(CommonJobProperties.EXECUTION_LINK,
           String.format("%s/executor?execid=%d", baseURL, this.executionId));
       this.props.put(CommonJobProperties.JOBEXEC_LINK, String.format(
-          "%s/executor?execid=%d&job=%s", baseURL, this.executionId, this.jobId));
+          "%s/executor?execid=%d&job=%s", baseURL, this.executionId, this.node.getNestedId()));
       this.props.put(CommonJobProperties.ATTEMPT_LINK, String.format(
           "%s/executor?execid=%d&job=%s&attempt=%d", baseURL, this.executionId,
-          this.jobId, this.node.getAttempt()));
+          this.node.getNestedId(), this.node.getAttempt()));
```
Comment (juhoautio) [inline azkaban-exec-server/src/main/java/azkaban/execapp/JobRunner.java]: Ok, nice. Looks like the UI is already using nestedId:

https://github.com/azkaban/azkaban/blob/4f30ad3b778ec6366251856158d4cc2dc68e7ad4/azkaban-web-server/src/web/js/azkaban/view/flow-execution-list.js#L391-L395

https://github.com/azkaban/azkaban/blob/4f30ad3b778ec6366251856158d4cc2dc68e7ad4/azkaban-web-server/src/web/js/azkaban/view/flow-execution-list.js#L215-L218

File: azkaban-exec-server/src/test/java/azkaban/execapp/FlowRunnerTest2.java [hunk 2]
```diff
@@ -162,6 +179,19 @@ public void testBasicRun() throws Exception {
...
+    assertThat(jobbInnerJobA.get(CommonJobProperties.EXECUTION_LINK))
+        .matches("http://localhost:8443/executor\\?execid=\\d\\d\\d");
+    assertThat(jobbInnerJobA.get(CommonJobProperties.WORKFLOW_LINK))
+        .matches("http://localhost:8443/manager\\?project=testProject&flow=innerFlow");
+    assertThat(jobbInnerJobA.get(CommonJobProperties.JOBEXEC_LINK))
+        .matches("http://localhost:8443/executor\\?execid=\\d\\d\\d&job=jobb:innerJobA");
+    assertThat(jobbInnerJobA.get(CommonJobProperties.JOB_LINK))
+        .matches("http://localhost:8443/manager\\?project=testProject&flow=innerFlow&job=innerJobA");
+    assertThat(jobbInnerJobA.get(CommonJobProperties.ATTEMPT_LINK))
+        .matches("http://localhost:8443/executor\\?execid=\\d\\d\\d&job=jobb:innerJobA&attempt=0");
```
Comment (juhoautio) [inline azkaban-exec-server/src/test/java/azkaban/execapp/FlowRunnerTest2.java]: Before the fix the link had `&job=innerJobA` and now it's `&job=jobb:innerJobA`, am I right?

File: azkaban-exec-server/src/test/java/azkaban/execapp/FlowRunnerTest2.java [hunk 3]
```diff
@@ -118,6 +123,18 @@ public void testBasicRun() throws Exception {
     assertEquals("test2.6", joba.get("param6"));
     assertEquals("test2.7", joba.get("param7"));
     assertEquals("test2.8", joba.get("param8"));
+    assertThat(joba.get(CommonJobProperties.JOB_ID)).isEqualTo("joba");
```
Comment (juhoautio) [inline azkaban-exec-server/src/test/java/azkaban/execapp/FlowRunnerTest2.java]: Just curious, any reason to not use plain `assertEquals`?

**Comment by juhoautio** [comment]
@burgerkingeater & co., this looks good to me

File: azkaban-exec-server/src/test/java/azkaban/execapp/FlowRunnerTest2.java (hunk 2 above)
Comment (edwinalu) [reply azkaban-exec-server/src/test/java/azkaban/execapp/FlowRunnerTest2.java]: Yes, this is the change.

File: azkaban-exec-server/src/test/java/azkaban/execapp/FlowRunnerTest2.java (hunk 3 above)
Comment (edwinalu) [reply azkaban-exec-server/src/test/java/azkaban/execapp/FlowRunnerTest2.java]: We're trying to use Assertj for newer tests: https://github.com/azkaban/azkaban/blob/master/CONTRIBUTING.md.

**Pull request merged**